### ISSUE TITLE
TypeScript Error: Movement is not part of state

### ISSUE THREAD

**Issue by do-kevin** [pmndrs/use-gesture#501]
**Information:**
 - Use Gesture version: ^10.2.4
 - TypeScript

**Bug**
I am getting an error complaining about state's movement. I've tried removing my node_modules and upgrading the package, but it still shows up. `movement` is still part of `state` on my console.log.

On hover, it says

```
Property 'movement' does not exist on type 'Omit<SharedGestureState & CommonGestureState & { axis: "x" | "y" | undefined; xy: Vector2; } & { _pointerId?: number | undefined; ... 7 more ...; swipe: Vector2; } & { ...; }, "event"> & { ...; }'.
```

**Screenshots**
![gesture linting bug](https://user-images.githubusercontent.com/40156784/171534657-b8d425b3-6ad5-4180-974d-0a44b601a135.PNG)
![console](https://user-images.githubusercontent.com/40156784/171534612-9685466b-0d12-4f38-8c8d-4d27040f76fe.PNG)

**Comment by linonetwo** [comment]
![截屏2022-07-06 23 25 44](https://user-images.githubusercontent.com/3746270/177586941-b1ee6886-d87d-4b45-83f2-697d458bd8c3.png)
![截屏2022-07-06 23 26 18](https://user-images.githubusercontent.com/3746270/177587032-957af225-e89e-49f0-a134-49da01d47c75.png)

**Comment by dbismut** [comment]
I won't reopen without a sandbox.

**Comment by Cuppachino** [comment]
I was able to recreate the issue by selecting "Use VS Code's Version 4.8.0-dev.20220714" over the workspace version, which in my case was 4.7.4

in the status bar, left of your selected language mode, click the curly braces.
![image](https://user-images.githubusercontent.com/92656523/179246936-493f0596-79b1-4515-b2fb-471ac9ff9589.png)

![image](https://user-images.githubusercontent.com/92656523/179246048-c6aa6e50-1557-4d6d-b94b-06197e54887a.png)

The fix is to your workspace version.

**Comment by dbismut** [comment]
@Cuppachino seems like TS 4.8 does introduce this bug. I'm not sure why tbh.

<img width="796" alt="image" src="https://user-images.githubusercontent.com/5003380/187079092-9fd6ee88-d662-4773-a42c-ecdbdd608890.png">

**Comment by dbismut** [comment]
[Here's a minimal repro on TS Playground.](https://www.typescriptlang.org/play?ts=4.8.0-beta#code/C4TwDgpgBA4hDOwCuAnCBpCIoF4oHIATFAQwHN8oAfAsASwDsBjAC3wCh3RIoBhAewC2g-gwDKwEsGh4A3lBEA3CIIgNgALigBtBkkEAjCCgA0UPYeMBdKAF8u4aABFSZCVJl8hI8ZOlQAMih5QjpEEmYILV19I1NzWOs7Bx4ABUZWd388AWFRLOgg+QiyABsohMsUZPZGaRQAMxImaALg9igoYnIAfi0XcgKOqHpmFj6odLGh+xToADEkUtK4RFQIAoAeTGwIAA9pBkJ4WARkNB2APlwoADlRW6XSkgNyzYLtHatLzm5oAAkIoRyihtlgoPtDsdTmsLlhrngABTDcLSLQAeUEdGAm0Wy1W5w2fggYJAlzM+AgynU+B+AEpcNcItgaIp+HRCJwmKJEFAAO78FAAaxOSNRFUx2NxTwJ6y2RFctIpVLUwFpDJw11kw25DHg-HKADpSvwyIjxYalCpVXT2LNdbzCPwEOoAOqCoVaQFHEGbBXkWk3c3EjVanU8g0QY2m4MeS38ZSqdS22xAA)

**Comment by dbismut** [comment]
Ok after some investigation, this is because of the simplification of the NonNullable mapped type.

```ts
// Native NonNullable Utility Type

// definition on 4.7 and lower
type NonNullable<T> = T extends null | undefined ? never : T

// definition from 4.8 
type NonNullable<T> = T & {}
```

Fix is incoming.

**Comment by dbismut** [comment]
Should be now fixed in 10.2.19.
